icu4c/source/common/loclikely.cpp

```diff
@@ -1379,11 +1379,10 @@ ulocimp_getRegionForSupplementalData(const char *localeID, UBool inferRegion,
         rgLen = 0;
     } else {
         // rgBuf guaranteed to be zero terminated here, with text len 6
-        char *rgPtr = rgBuf;
-        for (; *rgPtr!= 0; rgPtr++) {
-            *rgPtr = uprv_toupper(*rgPtr);
-        }
-        rgLen = (uprv_strcmp(rgBuf+2, "ZZZZ") == 0)? 2: 0;
+        // chop off the subdivision code (which will generally be "zzzz" anyway)
+        rgLen = 2;
+        rgBuf[0] = uprv_toupper(rgBuf[0]);
+        rgBuf[1] = uprv_toupper(rgBuf[1]);
     }
```

Comment:
And it does look like 3-digit regions are not allowed here (just confirming): https://www.unicode.org/reports/tr35/#RegionOverride

Reply:
It isn't currently. And I hope we never face it. 

But there is provision in BCP47 to require use of 3-digit codes if ISO reuses 2-letter codes. We'd have to change the spec at that point, but I wouldn't want ICU to blow up.